internal/node/hash.go

```diff
@@ -0,0 +1,153 @@
+package node
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"sort"
+	"strings"
+)
+
+// MethodsResult contains all available rpc methods of node
+type MethodsResult struct {
+	Version int      `json:"version"`
+	Methods []string `json:"methods"`
+}
+
+// RPCMethodsResponse is rpc response od rpc_methods call
+type RPCMethodsResponse struct {
+	JSONRPC string        `json:"jsonrpc"`
+	Result  MethodsResult `json:"result"`
+	ID      int           `json:"id"`
+}
+
+// RPCChainTypeResponse is rpc response of system_chainType call
+type RPCChainTypeResponse struct {
+	JSONRPC string `json:"jsonrpc"`
+	Result  string `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// RPCChainResponse is rpc response of system_chain call
+type RPCChainResponse struct {
+	JSONRPC string `json:"jsonrpc"`
+	Result  string `json:"result"`
+	ID      int    `json:"id"`
+}
+
+// RPCNodeRolesResponse is rpc response of system_nodeRoles call
+type RPCNodeRolesResponse struct {
+	JSONRPC string   `json:"jsonrpc"`
+	Result  []string `json:"result"`
+	ID      int      `json:"id"`
+}
+
+// Properties are node properties
+type Properties struct {
+	SS58Format    int    `json:"ss58Format"`
+	TokenDecimals int    `json:"tokenDecimals"`
+	TokenSymbol   string `json:"tokenSymbol"`
+}
+
+// RPCPropertiesResponse is rpc response of system_properties call
+type RPCPropertiesResponse struct {
+	JSONRPC string     `json:"jsonrpc"`
+	Result  Properties `json:"result"`
+	ID      int        `json:"id"`
+}
+
+// GetConfigHash hashes sorted node configuration
+func (client *client) GetConfigHash() (hash.Hash32, error) {
+	methods, err := client.getNodeRPCMethods()
+	if err != nil {
+		return nil, err
+	}
+
+	nodeRoles, err := client.getNodeRoles()
+	if err != nil {
+		return nil, err
+	}
+
+	chain, err := client.getChain()
+	if err != nil {
+		return nil, err
+	}
+
+	chainType, err := client.getChainType()
+	if err != nil {
+		return nil, err
+	}
+
+	properties, err := client.getNodeProperties()
+	if err != nil {
+		return nil, err
+	}
+
+	hash := fnv.New32()
+	sort.Strings(methods)
+	sort.Strings(nodeRoles)
+	_, _ = hash.Write([]byte(strings.Join(methods, "")))
+	_, _ = hash.Write([]byte(strings.Join(nodeRoles, "")))
+	_, _ = hash.Write([]byte(chain))
+	_, _ = hash.Write([]byte(chainType))
+	_, _ = hash.Write([]byte(fmt.Sprint(properties.SS58Format)))
+	_, _ = hash.Write([]byte(fmt.Sprint(properties.TokenDecimals)))
+	_, _ = hash.Write([]byte(string(properties.TokenSymbol)))
```

Comment:
sort config properties alphabetically. Examle chain should be before methods etc